quickstarts/microsoft.dbformysql/create-flexible-server-public-access/azuredeploy.json

```diff
@@ -0,0 +1,161 @@
+        },
+        "storageSizeGB": {
+            "type": "int"
+        },
+        "storageIops": {
+            "type": "int"
+        },
+        "storageAutogrow": {
+            "type": "string",
+            "defaultValue": "Enabled"
+        },
+        "availabilityZone": {
+            "type": "string",
+            "metadata": {
+                "description": "Availability Zone information of the server. (Leave blank for No Preference)."
+            }
+        },
+        "version": {
+            "type": "string"
+        },
+        "tags": {
+            "type": "object",
+            "defaultValue": {}
+        },
+        "haEnabled": {
+            "type": "string",
+            "defaultValue": "Disabled",
+            "metadata": {
+                "description": "High availability mode for a server : Disabled, SameZone, or ZoneRedundant"
+            }
+        },
+        "standbyAvailabilityZone": {
+            "type": "string",
+            "metadata": {
+                "description": "Availability zone of the standby server."
+            }
+        },
+        "firewallRules": {
+            "type": "object",
+            "defaultValue": {}
+        },
+        "backupRetentionDays": {
+            "type": "int"
+        },
+        "geoRedundantBackup": {
+            "type": "string"
+        },
+        "databaseName": {
+            "type": "string"
+        }
+    },
+    "variables": {
+        "api": "2021-05-01",
+        "firewallRules": "[parameters('firewallRules').rules]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.DBforMySQL/flexibleServers",
+            "apiVersion": "[variables('api')]",
+            "location": "[parameters('location')]",
+            "name": "[parameters('serverName')]",
+            "sku": {
+                "name": "[parameters('skuName')]",
+                "tier": "[parameters('serverEdition')]"
+            },
+            "properties": {
+                "version": "[parameters('version')]",
+                "administratorLogin": "[parameters('administratorLogin')]",
+                "administratorLoginPassword": "[parameters('administratorLoginPassword')]",
+                "availabilityZone": "[parameters('availabilityZone')]",
+                "highAvailability": {
+                    "mode": "[parameters('haEnabled')]",
+                    "standbyAvailabilityZone": "[parameters('standbyAvailabilityZone')]"
+                },
+                "Storage": {
+                    "storageSizeGB": "[parameters('storageSizeGB')]",
+                    "iops": "[parameters('storageIops')]",
+                    "autogrow": "[parameters('storageAutogrow')]"
+                },
+                "Backup": {
+                    "backupRetentionDays": "[parameters('backupRetentionDays')]",
+                    "geoRedundantBackup": "[parameters('geoRedundantBackup')]"
+                } 
+            },
+            "tags": "[parameters('tags')]"  
```

Comment:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties